src/it/scala/constellation/IntegrationSpec.scala

```diff
@@ -71,25 +65,48 @@ class IntegrationSpec extends FunSpec with Matchers with BeforeAndAfterAll {
 
     def checkWhileReady: IO[String] = checkIfReady.flatMap {
       case a if a == "2" => IO.pure(a)
-      case o => IO.sleep(100 millis).flatMap(_ => checkWhileReady)
+      case o             => IO.sleep(100 millis).flatMap(_ => checkWhileReady)
     }
 
+    def enableMaintenance: IO[Unit] =
+      httpClient.use {
+        _.expect[Unit](Request[IO](method = POST, uri = uri("http://localhost:9001/settings/maintenance")))
+      }
+
     checkWhileReady.unsafeRunSync()
 
+    it("Returns 503 when maintenance mode enabled") {
+      enableMaintenance.unsafeRunSync
+      val result = httpClient.use {
+        _.status(Request[IO](method = POST, uri = uri("http://localhost:9000/dashboard")))
+      }.unsafeRunSync
+
+      result should equal(Status.ServiceUnavailable)
```

Comment:
I wonder should we maybe assert on the body also? So that we are sure that the response comes from the part of code we expect it should come from?

Reply:
@TheMMaciek if there must be a specific body - let's check it
if the expected output is a 503 status only, then there is no point for adding additional body just to check it in test

Reply:
The body is there, and it's indicating the reason why service is unavailable. If there will be another functionality or some default behavior returning ServiceUnavailable this test could still pass.

Reply:
Imo status code is enough as we have no multiple 503 reasons in the app for now